### PR TITLE
[DO NOT MERGE] Update script for a Rails 7 app with `importmap`

### PR DESCRIPTION
**Pull request by Annedj**
## Context

As part of the `More JS in web dev` project, we will now be using `importmaps`, so we need to update all our Rails templates and apps.

- [x] Update the `minimal` template
- [x] Update the `devise` template

The minimal template seems to work fine, the only thing you need to do after creating an app with it:
```bash
rails new \                            
  -d postgresql \
  -m https://raw.githubusercontent.com/lewagon/rails-templates/import-maps/minimal.rb \
  name_of_the_app
```

is to change line 7 in `application.scss`: (I added it in the script)
```scss
// @import "bootstrap/scss/bootstrap";
// becomes:
@import "bootstrap";
```

I'm continuing with `devise` template now but let me know if you're having issues with the first one 🙏

### CODE REVIEW

File: minimal.rb [hunk 1]
```diff
@@ -4,32 +4,31 @@
 ########################################
 inject_into_file "Gemfile", before: "group :development, :test do" do
   <<~RUBY
+    gem "bootstrap"
```
Comment (trandthanh) [inline minimal.rb]: Should we fix the version?

File: minimal.rb [hunk 2]
```diff
@@ -4,32 +4,31 @@
...
+  "\n  gem \"dotenv-rails\""
 end
 
 gsub_file("Gemfile", '# gem "sassc-rails"', 'gem "sassc-rails"')
 
 # Assets
 ########################################
 run "rm -rf app/assets/stylesheets"
-run "rm -rf vendor"
+# run "rm -rf vendor"
```
Comment (trandthanh) [inline minimal.rb]: We need the vendor folder now? 🤔

File: minimal.rb [hunk 3]
```diff
@@ -109,4 +110,5 @@
   git :init
   git add: "."
   git commit: "-m 'Initial commit with minimal template from https://github.com/lewagon/rails-templates'"
+  # git branch: "-m main master"
```
Comment (trandthanh) [inline minimal.rb]: I think students are using `main` now no?

File: minimal.rb [hunk 4]
```diff
@@ -4,32 +4,31 @@
...
-run "rm -rf vendor"
+# run "rm -rf vendor"
 run "curl -L https://github.com/lewagon/rails-stylesheets/archive/master.zip > stylesheets.zip"
 run "unzip stylesheets.zip -d app/assets && rm -f stylesheets.zip && rm -f app/assets/rails-stylesheets-master/README.md"
 run "mv app/assets/rails-stylesheets-master app/assets/stylesheets"
 
 inject_into_file "config/initializers/assets.rb", before: "# Precompile additional assets." do
   <<~RUBY
-    Rails.application.config.assets.paths << Rails.root.join("node_modules")
+    Rails.application.config.assets.paths << Rails.root.join("bootstrap.min.js popper.js")
```
Comment (trandthanh) [inline minimal.rb]: 👍

**Review by trandthanh — COMMENTED** [review]
🚀 ⚡ 🙏 

Did you check using Bootstrap JS in the app? 🙏

File: minimal.rb (hunk 3 above)
Comment (Annedj) [reply minimal.rb]: Ah I wondered!

**Comment by Annedj** [comment]
Bootstrap JS wasn't working properly so I found another way but it involves adding both popper and bootstrap in the `manifest.js, it works now but I feel like that's not the way to go... 

Cf this [article](https://dev.to/coorasse/rails-7-bootstrap-5-and-importmaps-without-nodejs-4g8) but I ended up with a `ManifestError` so I had to add it there.

**Comment by Annedj** [comment]
I think we are good for the devise template too:

image

**Comment by trandthanh** [comment]
Based on our investigation earlier, if you pin bootstrap via the `importmap` command in the terminal, does it import the popper dependency? Just curious 🤔.

**Comment by Annedj** [comment]
If you use the `importmap` command, it adds these two lines:
```rb
pin "bootstrap", to: "https://ga.jspm.io/npm:bootstrap@5.2.3/dist/js/bootstrap.esm.js"
pin "@popperjs/core", to: "https://ga.jspm.io/npm:@popperjs/core@2.11.6/lib/index.js"
```

so I thought the other two were more readable, but at least pin to the specific version so that might be clearer! I can update the script if we prefer this 👌

**Comment by trandthanh** [comment]
Hello @Annedj 👋,

I think I like using the `importmap` binary. It's a cleaner approach, WDYT?

Also, could you change the README as well 😉 on this PR? I think it still mentions the `-j webpack` flag? 

Thank you 🙏

**Comment by Annedj** [comment]
Done ✅

File: devise.rb [hunk 5]
```diff
@@ -28,9 +27,11 @@
 run "unzip stylesheets.zip -d app/assets && rm -f stylesheets.zip && rm -f app/assets/rails-stylesheets-master/README.md"
 run "mv app/assets/rails-stylesheets-master app/assets/stylesheets"
 
+gsub_file("app/assets/stylesheets/application.scss", '@import "bootstrap/scss/bootstrap";', '@import "bootstrap";')
+
 inject_into_file "config/initializers/assets.rb", before: "# Precompile additional assets." do
   <<~RUBY
-    Rails.application.config.assets.paths << Rails.root.join("node_modules")
+    Rails.application.config.assets.paths << Rails.root.join("bootstrap.min.js popper.js")
```
Comment (Annedj) [inline devise.rb]: I feel like we don't actually need this ⚠️

Comment (trandthanh) [reply devise.rb]: Did you experiment without it? 🙏

Comment (Annedj) [reply devise.rb]: Nope still need to, that was a reminder to myself 😄

Comment (Annedj) [reply devise.rb]: Good news, it works without so I removed it from the script and corrected the `popper` package:

image

**Comment by ajdubovoy** [comment]
@juliends I just realized we should also merge this one, agreed? I can make a pr to remove references to the import-maps branch to master in karr/challenges

**Comment by juliends** [comment]
> @juliends I just realized we should also merge this one, agreed? I can make a pr to remove references to the import-maps branch to master in karr/challenges

Definitely 🙏

**Review by juliends — APPROVED** [review]

**Pull request merged**